shared/common-adapters/text.native.js

```diff
@@ -0,0 +1,271 @@
+      case 'HeaderJumbo':
+      case 'HeaderBig':
+      case 'Header':
+      case 'BodySemibold':
+      case 'BodySemiboldItalic':
+      case 'BodySmallSemibold':
+      case 'Body':
+        return {color: backgroundMode === 'Normal' ? globalColors.black75 : globalColors.white}
+      case 'BodySmall':
+        return {color: backgroundMode === 'Normal' ? globalColors.black40 : globalColors.white40}
+      case 'BodyPrimaryLink':
+        return {color: globalColors.blue}
+      case 'BodySecondaryLink':
+        return {color: globalColors.black40}
+      default:
+        return {}
+    }
+  }
+
+  focus () {
+    if (this.refs && this.refs.text) {
+      this.refs.text.focus()
+    }
+  }
+
+  render () {
+    const typeStyle = {
+      'HeaderJumbo': styles.textHeaderJumbo,
+      'HeaderBig': styles.textHeaderBig,
+      'Header': styles.textHeader,
+      'HeaderError': {...styles.textHeader, color: globalColors.red},
+      'BodySemibold': styles.textBodySemibold,
+      'BodySemiboldItalic': {...styles.textBodySemibold, ...globalStyles.italic},
+      'Body': styles.textBody,
+      'BodyPrimaryLink': styles.textBody,
+      'BodySecondaryLink': styles.textBodySmall,
+      'BodySmall': styles.textBodySmall,
+      'BodySmallSemibold': styles.textBodySmallSemibold,
+      'Error': styles.textError,
+      'Terminal': {...styles.textTerminal, color: (this.context.inTerminal ? globalColors.blue3 : globalColors.darkBlue)},
+      'TerminalSmall': {...styles.textTerminalSmall, color: (this.context.inTerminal ? globalColors.blue3 : globalColors.darkBlue)},
+      'TerminalCommand': styles.textTerminalCommand,
+      'TerminalComment': styles.textTerminalComment,
+      'TerminalUsername': styles.textTerminalUsername,
+      'TerminalPublic': styles.textTerminalPublic,
+      'TerminalPrivate': styles.textTerminalPrivate,
+      'TerminalEmpty': styles.textTerminalEmpty,
+      'InputHeader': styles.textInputHeader
+    }[this.props.type]
+
+    let inline = true
+    if (this.props.hasOwnProperty('inline')) {
+      inline = this.props.inline
+    }
+
+    let linkStyle = null
+    let linkClassname = null
+    switch (this.props.type) {
+      case 'BodyPrimaryLink':
+      case 'BodySecondaryLink':
+        linkStyle = {}
+        linkClassname = 'hover-underline'
+    }
+
+    const style = {
+      ...typeStyle,
+      ...linkStyle,
+      ...this._colorStyleBackgroundMode(this.props.backgroundMode || 'Normal', this.props.type),
+      ...(this.props.onClick ? globalStyles.clickable : {}),
+      ...(inline ? {...this._inlineStyle(this.props.type)} : {}),
+      ...this.props.style
+    }
+
+    const terminalPrefix = this._terminalPrefix(this.props.type)
+    const className = (this.props.className || '') + ' ' + (linkClassname || '')
+
+    if (this.props.contentEditable) {
+      return (
+        <RNText
+          ref='text'
+          className={className}
+          style={style}
+          contentEditable
```

Comment:
k, i fully expect us to have to re-touch these files as we expand out the rest of the components